File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetsbankingtransactioncategory.go
package shared

import (
	"time"
)

type CodatDataContractsDatasetsBankingTransactionCategory struct {
	HasChildren        *bool                                                           `json:"hasChildren,omitempty"`
	ID                 string                                                          `json:"id"`
	ModifiedDate       *time.Time                                                      `json:"modifiedDate,omitempty"`
	Name               string                                                          `json:"name"`
	ParentID           *string                                                         `json:"parentId,omitempty"`
	SourceModifiedDate *time.Time                                                      `json:"sourceModifiedDate,omitempty"`
	Status             *CodatDataContractsDatasetsBankingTransactionCategoryStatusEnum `json:"status,omitempty"`
}

File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_banking_transactioncategory.py
import dataclasses
import dateutil.parser
from ..shared import codat_datacontracts_datasets_banking_transactioncategorystatus_enum as shared_codat_datacontracts_datasets_banking_transactioncategorystatus_enum
from dataclasses_json import dataclass_json
from datetime import datetime
from marshmallow import fields
from sdk import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsBankingTransactionCategory:
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    has_children: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('hasChildren') }})
    modified_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('modifiedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    parent_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('parentId') }})
    source_modified_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sourceModifiedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    status: Optional[shared_codat_datacontracts_datasets_banking_transactioncategorystatus_enum.CodatDataContractsDatasetsBankingTransactionCategoryStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    